02.calendar/cal.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require 'date'
+require 'optparse'
+
+CALENDER_WIDTH = 20
+DAY_DIGIT = 2
+MIN_YEAR = 1
+MAX_YEAR = 9999
+
+def main
+  today = Date.today
+  opt = OptionParser.new
+  opt.on('-m')
+  opt.on('-y')
+  params = ARGV.getopts('', "y:#{today.year}", "m:#{today.month}")
+  year = params['y'].to_i
+  month = params['m'].to_i
+
+  unless valid_year?(year)
+    puts "cal: year `#{year}' not in range #{MIN_YEAR}..#{MAX_YEAR}"
+    return
+  end
+
+  unless valid_month?(month)
+    puts "cal: #{month} is neither a month number (1..12) nor a name"
+    return
+  end
+
+  calender_header(year, month)
+  dates = generate_dates(year, month)
+  calender_body(today, dates)
+end
+
+def valid_year?(year)
+  year >= MIN_YEAR && year <= MAX_YEAR
+end
+
+def valid_month?(month)
+  month >= 1 && month <= 12
```

Comment:
```suggestion
  (1..12).include?(month)
```

でもいいかも。yearの方も同様ですね。

Reply:
includeメソッドに変更しました！

Reply:
RuboCopにて`C: [Correctable] Performance/RangeInclude: Use Range#cover? instead of Range#include?.` が表示されたため、include?ではなくcover?に変更しました。数値のRangeオブジェクトに対する適用でinclude?と挙動が変わらないと考えたため変更しています。